### PR TITLE
Git hub auth

### PR DESCRIPTION
**Pull request by edopore**

### CODE REVIEW

**Comment by edopore** [comment]
Hi, I have an issue with CI, test failed, the error is based on the test part, I make PR when CI starts now the first tes about gihub authentication fails, but second test pass, I change PAT from github and probe with another ACCESS_TOKEN variable but make the same his line isn't on last branch, from my pc test pass but when I make PR and CI test begin this test fail, what may I do to solve this?

firstTestBad

problemtest2

**Comment by edopore** [comment]
codeeslinvscode

File: test/GithubApi.Authentication.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const { StatusCodes } = require('http-status-codes');
+const { expect } = require('chai');
+const axios = require('axios');
+
+const urlBase = 'https://api.github.com';
+const githubUserName = 'AgileTestingColombia';
```
Comment (holgiosalos) [inline test/GithubApi.Authentication.test.js]: please change this to be your user name `'edopore'`

**Review by holgiosalos — COMMENTED** [review]
If your test continues failing, I don't know why is failing

File: test/GithubApi.Authentication.test.js (hunk 1 above)
Comment (edopore) [reply test/GithubApi.Authentication.test.js]: I changed username, but test don't pass anyway

**Comment by holgiosalos** [comment]
I don't know why continues failing 😞

**Comment by veronicatofino** [comment]
Did you configure correctly the secret ACCESS_TOKEN variable in the repository? @edopore

**Review by holgiosalos — COMMENTED** [review]
@edopore could you check two things?

1. you have correctly created the token.
2. you have defined in your repo, the secret with key ACCESS_TOKEN and value, your generated token

**Comment by edopore** [comment]
> @edopore could you check two things?
> 
> 1. you have correctly created the token.
> 2. you have defined in your repo, the secret with key ACCESS_TOKEN and value, your generated token

Yes, in fact I made a new Personal Access Token and Modify the value of ACCESS_TOKEN variable with this new PAT I surprise that first test fail but second test pass. Somewhere in stack overflow forum I read the same problem shows bad request that later pass the test.

**Comment by edopore** [comment]
I enable debug test on CI github ACCESS_TOKEN shows as null. Why this could be?

7BE12CE8-42EC-410B-A69E-512E23ABDC78

**Comment by edopore** [comment]
I found an error about environment, initially when I added ACCESS_TOKEN Variable GitHub send me to a secret environmet, something different to repository secrets, tha was my mistake, I just put ACCESS_TOKEN variable in repository secret.
Now I have a doubt about differences between Secret Environment and Repository Secrets.

**Comment by holgiosalos** [comment]
@edopore an environment is a place where an application can be deployed, environment secrets, are secrets that only apply to a specific environment, on the other side, the repository secrets apply to the whole repository and all the things related to the repository like the github actions. We are not handling environments in this workshop

**Review by holgiosalos — APPROVED** [review]

**Pull request merged**